### PR TITLE
Enh/prepare dataset 28

### LINKED ISSUE #28: Prepare dataset
## Write methods to prepare the dataset
Write a method to extract the images out of the file, cut out the faces, scales the images and saves them in another folder.

Write a methode that splits the dataset in training, validation and test set.

Nice2Have:
Method that rotates the images and save multiple versions with differnet angles or in other ways creates a bit different versions off the picture

### PR DESCRIPTION
**Pull request by Paeti**
closes #28

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -102,3 +102,4 @@ venv.bak/
 
 # mypy
 .mypy_cache/
+.idea
```
Comment (frzifus) [inline .gitignore]: shouldn't we put the folder on the `.gitignore` for all sub folders too?
if I remembered it correctly, last time the `mains/.idea` folder was uploaded.

**Comment by frzifus** [comment]
Also i don't think it's a good idea to keep old files as marked `alt`.
`haarcascade_frontalface_alt2.xml`,`haarcascade_frontalface_alt.xml`, `haarcascade_frontalface_alt_tree.xml`, ...
That's why git is available as version management system.

File: mains/parse_metadata.py [hunk 2]
```diff
@@ -98,3 +100,34 @@
...
+#convert data to features
+def _int64_feature(value):
+    return tf.train.Feature(int64_list=tf.train.Int64List(value=[value]))
+
+def _bytes_feature(value):
+    return tf.train.Feature(bytes_list=tf.train.BytesList(value=[value]))
+
+
+#write data in tfrecords file
+train_filename = 'train.tfrecords'
```
Comment (frzifus) [inline mains/parse_metadata.py]: (maybe you should just execute this piece of code explicitly.
)[https://stackoverflow.com/questions/419163/what-does-if-name-main-do]
```python
if __name__ == "__main__":
```

File: mains/parse_metadata.py [hunk 3]
```diff
@@ -98,3 +100,34 @@
...
+def load_image(addr):
+    img = cv.imread(addr)
+    #img = cv.resize(img, (224, 224), interpolation=cv.INTER_CUBIC)
+    img = cv.cvtColor(img, cv.COLOR_BGR2RGB)
+    img = img.astype(np.float32)
+    return img
+
+
+#convert data to features
+def _int64_feature(value):
```
Comment (frzifus) [inline mains/parse_metadata.py]: maybe you should choose a different function name here.
https://stackoverflow.com/questions/1301346/what-is-the-meaning-of-a-single-and-a-double-underscore-before-an-object-name

File: mains/parse_metadata.py [hunk 4]
```diff
@@ -98,3 +100,34 @@
...
+    return tf.train.Feature(int64_list=tf.train.Int64List(value=[value]))
+
+def _bytes_feature(value):
```
Comment (frzifus) [inline mains/parse_metadata.py]: maybe you should choose a different function name here.

**Comment by Paeti** [comment]
mkdir preprocessing
mv IPNeuronaleNetze/mains/* IPNeuronaleNetze/preprocessing/
.gitignore => .idea folder is located in mains => in .gitignore change .idea to mains/.idea
change *_alt.py filenames in selfexplaining filenames
the project structure uses object orientation => use classes

File: mains/parse_metadata.py (hunk 2 above)
Comment (DZvO) [reply mains/parse_metadata.py]: The script is used only once to train the initial network, I doubt we'll ever call it from anywhere else except maybe a setup bash script or something like that.

File: mains/parse_metadata.py (hunk 3 above)
Comment (DZvO) [reply mains/parse_metadata.py]: Just copy-pasted from a tutorial, again: only used once no need to apply OO-Patterns IMHO.
